index.d.ts

```diff
@@ -0,0 +1,166 @@
+// Type definitions for jQuery-QueryBuilder v2.4.3
+// Definitions by: Ron Newcomb <https://github.com/RonNewcomb>
+
+/// <reference types="jquery" />
+
+declare namespace jQueryQueryBuilder {
+
+    interface QueryBuilder {
+        constructor($el: JQuery, options: Options): QueryBuilder;
+        //static readonly DEFAULTS: object;
+        //static Group: Function;
+        //static readonly OPERATORS : Map<string, object>;
+        //static readonly regional: Map<string, object>;
+        //static Rule: Function;
+        //static readonly selectors: Map<string, string>;
+        //static readonly templates: Map<string, string>;
+        //static utils: object;
+        readonly $el: JQuery;
+        readonly filters: Array<FilterOptions>;
+        readonly icons: {};
+        readonly lang: object;
+        //readonly model: Model;
+        readonly operators: Array<OperatorDefinition>;
+        readonly plugins: Map<string, object>;
+        readonly settings: object;
+        readonly templates: Map<string, Function>;
+    }
+
+    interface QueryBuilderEvent extends Event {
+        builder: QueryBuilder;
+    }
+
+    interface TriggerHandler {
+        <T>(e: QueryBuilderEvent, rule: Rule, error: string, value: T): void;
+    }
+
+    interface ChangerHandler {
+        <T>(e: QueryBuilderEvent & { value: T }, level: number): void;
+    }
+
+    interface Conjunction {
+        condition: string;
+        rules: Rule[];
```

Comment:
Actually it's not a Rule object but a plain Json, it should be a new RuleJson type